blinkit-app/src/homePage/organisms/HomePage.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+import vegetables, { total } from "../../DATA/ITEMS";
+import { UserProvider } from "../../userContext";
+import Foodbar from "../molecules/foodbar/Foodbar";
+import Footer from "../molecules/footer/Footer";
+import Header from "../molecules/header/Header";
+import MainPage from "../molecules/mainPage/MainPage";
+import Speciality from "../molecules/specialities/Speciality";
+import Information from "../molecules/information/Information"
+
+class HomePage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: vegetables,
+      total: total,
+      query: "",
+      queryOutput: vegetables,
+    };
+  }
+  add = (index) => {
+    console.log(index);
+    vegetables[index].quantity = vegetables[index].quantity + 1;
+    total.item += 1;
+    total.price += Number(vegetables[index].currentPrice);
+    this.setState({
+      items: vegetables,
+      total: total,
+    });
+  };
+  minus = (index) => {
+    vegetables[index].quantity = vegetables[index].quantity - 1;
+    total.item -= 1;
+    total.price -= Number(vegetables[index].currentPrice);
+    this.setState({
+      items: vegetables,
+    });
+  };
+
+  queryHandler = (e) => {
+    const result = this.state.items.filter((item) => {
+      if (e.target.value === "") return this.state.items;
+      return item.name.toLowerCase().includes(e.target.value.toLowerCase());
+    });
+    console.log(e.target.value);
+    console.log(result);
+    this.setState({
+      queryOutput: result,
+    });
+  };
+  filterHandler = (e) => {
+    const result = this.state.items.filter((item) => {
+      if (e.target.value === "Relevance") return this.state.items;
+      else if (e.target.value === "Veg Only")
```

Comment:
Ids are written in camelCase.

Second,
While comparing ids we need to main a single source of truth. So create an array of options, which will have the name and the id. Use that object id to compare it here and the array to create options.

```
const VEG_ONLY={
id: 'vegOnly',
name: 'Veg Only'
}
const FILTER_OPTIONS=[VEG_ONLY,....];
if(event.target.value===VEG_ONLY.id){....}
```